metaverse_modules/nameplate/model.js

```diff
@@ -0,0 +1,64 @@
+import {Box3} from 'three';
+import {Geometry} from 'three/examples/jsm/deprecated/Geometry';
+
+const addToMaterials = (materials, newMaterial) => {
+  for (let i = 0; i < materials.length; i++) {
+    const mat = materials[i];
+    if (mat.name === newMaterial.name) {
+      return [materials, i];
+    }
+  }
+  materials.push(newMaterial);
+  return [materials, materials.length - 1];
+};
+
+export const getModelGeoMat = model => {
+  const newGeometry = new Geometry();
+  let newMaterials = [];
+  model.scene.traverse(function(child) {
+    if (child.type === 'Mesh') {
+      const materialIndices = [];
+      let newItems;
+      if (child.material.length) {
+        for (let k = 0; k < child.material.length; k++) {
+          newItems = addToMaterials(newMaterials, child.material[k]);
+          newMaterials = newItems[0];
+          materialIndices.push(newItems[1]);
+        }
+      } else {
+        newItems = addToMaterials(newMaterials, child.material); // materials.push(child.material);
+        newMaterials = newItems[0];
+        materialIndices.push(newItems[1]);
+      }
+
+      if (child.geometry.isBufferGeometry) {
+        const tGeometry = new Geometry().fromBufferGeometry(child.geometry);
```

Comment:
Why use a non-buffer geometry?

Reply:
I had to use legacy geometry to keep children's position when merging several meshes.